src/main/scala/org/constellation/primitives/EdgeDAO.scala

```diff
@@ -83,6 +78,19 @@ class ThreadSafeTipService() {
     acceptedCBSinceSnapshot = acceptedCBSinceSnapshot.filterNot(snapshot.checkpointBlocks.contains)
   }
 
+  // TODO: Read from lastSnapshot in DB optionally, assign elsewhere
+  var lastSnapshotHeight = 0
+
+  def getMinTipHeight()(implicit dao: DAO) = thresholdMetCheckpoints.keys.map {
+    dao.checkpointService.get
+  }.flatMap {
+    _.flatMap {
+      _.height.map {
+        _.min
```

Comment:
this is pretty gnarly. This inner flatMap / flatMap / map seems like a great use of a for comprehension

Reply:
Will fix